protostar/argument_parser/unparser.py

```diff
@@ -0,0 +1,30 @@
+from typing import Any, Optional
+
+
+def unparse_arguments_from_external_source(value: Any) -> Optional[list[str]]:
+    """Arguments from external sources need to be unparsed in order to be parsed by custom parsers."""
+    if value is None:
+        return None
+    if isinstance(value, bool):
+        return None
+    return unparse_arguments(value)
+
+
+def unparse_arguments(value: Any) -> list[str]:
+    if not isinstance(value, list):
+        return [unparse_single_value(value)]
+
+    values = value
+    unparsed_values: list[str] = []
+    for val in values:
+        unparsed_values.append(*unparse_arguments(val))
```

Comment:
This can be replaced with list comprehension

Reply:
> Pylance: Unpack operation not allowed in comprehension